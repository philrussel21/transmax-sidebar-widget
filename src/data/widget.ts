import {IconType, Status} from 'src/library/types';

const data = {
  weather: {
    label: 'Melbourne',
    temperature: 32,
    dateTime: new Date().toISOString(),
    icon: 'snowy' as IconType,
  },
  forecast: [
    {
      label: 'Humidity',
      value: '78%',
    },
    {
      label: 'Chance of Rain',
      value: '34%',
    },
    {
      label: 'Wind',
      value: '21',
      meta: 'kmh',
    },
    {
      label: 'Tomorrow',
      value: '30Â°',
      icon: 'sunny' as IconType,
    },
  ],
  delayedRoutes: [
    {
      id: '1',
      status: 'high' as Status,
      name: 'Monash Fwy Out',
      via: ['Kings Way', 'Eastlink'],
      distance: 13,
      duration: 45,
    },
    {
      id: '2',
      status: 'high' as Status,
      name: 'Monash Fwy Out',
      via: ['Kings Way', 'Eastlink'],
      distance: 15,
      duration: 28,
    },
    {
      id: '3',
      status: 'medium' as Status,
      name: 'Western Ring Road',
      via: ['West Gate Fwy', 'Western Fwy'],
      distance: 5,
      duration: 5,
    },
    {
      id: '4',
      status: 'medium' as Status,
      name: 'Eastern Fwy',
      via: ['Hoddle St', 'Springvale Rd'],
      distance: 15,
      duration: 25,
    },
  ],
};

export default data;
