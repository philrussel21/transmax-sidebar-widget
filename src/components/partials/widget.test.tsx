import {render} from '@testing-library/react';
import Widget from './widget';
import data from 'src/data/widget';

test('snapshot', () => {
  const result = render(<Widget />);

  expect(result).toMatchSnapshot();
});

test('it should show weather data by default', () => {
  const {getByText} = render(<Widget />);

  expect(getByText(data.weather.label)).toBeInTheDocument();
});

test('it should not show delayed routes by default', () => {
  const {queryByText} = render(<Widget />);

  expect(queryByText(data.delayedRoutes[0].name)).toBeNull();
  expect(queryByText(data.delayedRoutes[0].duration)).toBeNull();
  expect(queryByText(data.delayedRoutes[0].distance)).toBeNull();
});
