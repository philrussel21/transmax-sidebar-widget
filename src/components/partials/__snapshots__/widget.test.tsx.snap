// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-4 bg-[#15191E]"
      >
        <div
          class="px-4"
        >
          <div
            class="flex gap-4 justify-between items-center font-light"
          >
            <div>
              <h4
                class="ml-1 text-sm"
              >
                Melbourne
              </h4>
              <span
                class="text-5xl ml-3"
              >
                32
                °
              </span>
              <div
                class="flex text-xs gap-2"
              >
                <span>
                  Sun 7th
                </span>
                <span>
                  9:29 PM
                </span>
              </div>
            </div>
            <div>
              <svg
                class="size-20"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M440.1 355.2l-39.2-23 34.1-9.3c8.4-2.3 13.4-11.1 11.1-19.6l-4.1-15.5c-2.2-8.5-10.9-13.6-19.3-11.3L343 298.2 271.2 256l71.9-42.2 79.7 21.7c8.4 2.3 17-2.8 19.3-11.3l4.1-15.5c2.2-8.5-2.7-17.3-11.1-19.6l-34.1-9.3 39.2-23c7.5-4.4 10.1-14.2 5.8-21.9l-7.9-13.9c-4.3-7.7-14-10.3-21.5-5.9l-39.2 23 9.1-34.7c2.2-8.5-2.7-17.3-11.1-19.6l-15.2-4.1c-8.4-2.3-17 2.8-19.3 11.3l-21.3 81-71.9 42.2v-84.5L306 70.4c6.1-6.2 6.1-16.4 0-22.6l-11.1-11.3c-6.1-6.2-16.1-6.2-22.2 0l-24.9 25.4V16c0-8.8-7-16-15.7-16h-15.7c-8.7 0-15.7 7.2-15.7 16v46.1l-24.9-25.4c-6.1-6.2-16.1-6.2-22.2 0L142.1 48c-6.1 6.2-6.1 16.4 0 22.6l58.3 59.3v84.5l-71.9-42.2-21.3-81c-2.2-8.5-10.9-13.6-19.3-11.3L72.7 84c-8.4 2.3-13.4 11.1-11.1 19.6l9.1 34.7-39.2-23c-7.5-4.4-17.1-1.8-21.5 5.9l-7.9 13.9c-4.3 7.7-1.8 17.4 5.8 21.9l39.2 23-34.1 9.1c-8.4 2.3-13.4 11.1-11.1 19.6L6 224.2c2.2 8.5 10.9 13.6 19.3 11.3l79.7-21.7 71.9 42.2-71.9 42.2-79.7-21.7c-8.4-2.3-17 2.8-19.3 11.3l-4.1 15.5c-2.2 8.5 2.7 17.3 11.1 19.6l34.1 9.3-39.2 23c-7.5 4.4-10.1 14.2-5.8 21.9L10 391c4.3 7.7 14 10.3 21.5 5.9l39.2-23-9.1 34.7c-2.2 8.5 2.7 17.3 11.1 19.6l15.2 4.1c8.4 2.3 17-2.8 19.3-11.3l21.3-81 71.9-42.2v84.5l-58.3 59.3c-6.1 6.2-6.1 16.4 0 22.6l11.1 11.3c6.1 6.2 16.1 6.2 22.2 0l24.9-25.4V496c0 8.8 7 16 15.7 16h15.7c8.7 0 15.7-7.2 15.7-16v-46.1l24.9 25.4c6.1 6.2 16.1 6.2 22.2 0l11.1-11.3c6.1-6.2 6.1-16.4 0-22.6l-58.3-59.3v-84.5l71.9 42.2 21.3 81c2.2 8.5 10.9 13.6 19.3 11.3L375 428c8.4-2.3 13.4-11.1 11.1-19.6l-9.1-34.7 39.2 23c7.5 4.4 17.1 1.8 21.5-5.9l7.9-13.9c4.6-7.5 2.1-17.3-5.5-21.7z"
                />
              </svg>
            </div>
          </div>
          <div
            class="mt-5"
          >
            <ul
              class="space-y-0.5"
            >
              <li
                class="flex gap-5 text-sm items-center"
              >
                <span
                  class="font-light flex-1"
                >
                  Humidity
                </span>
                <span
                  class="flex gap-0.5 items-center w-1/3"
                >
                  <span
                    class="font-medium text-base"
                  >
                    78%
                  </span>
                </span>
              </li>
              <li
                class="flex gap-5 text-sm items-center"
              >
                <span
                  class="font-light flex-1"
                >
                  Chance of Rain
                </span>
                <span
                  class="flex gap-0.5 items-center w-1/3"
                >
                  <span
                    class="font-medium text-base"
                  >
                    34%
                  </span>
                </span>
              </li>
              <li
                class="flex gap-5 text-sm items-center"
              >
                <span
                  class="font-light flex-1"
                >
                  Wind
                </span>
                <span
                  class="flex gap-0.5 items-center w-1/3"
                >
                  <span
                    class="font-medium text-base"
                  >
                    21
                  </span>
                  <span
                    class="font-light"
                  >
                    kmh
                  </span>
                </span>
              </li>
              <li
                class="flex gap-5 text-sm items-center"
              >
                <span
                  class="font-light flex-1"
                >
                  Tomorrow
                </span>
                <span
                  class="flex gap-0.5 items-center w-1/3"
                >
                  <span
                    class="font-medium text-base"
                  >
                    30°
                  </span>
                  <svg
                    class="size-4"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
                    />
                  </svg>
                </span>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="mt-8"
        >
          <section
            class="space-y-4"
          >
            <article
              class="rounded-t-sm bg-[#222931]"
            >
              <header>
                <button
                  class="w-full flex justify-between items-center px-4 py-2 border border-white/10"
                  type="button"
                >
                  <span
                    class="uppercase"
                  >
                    Delayed Routes
                  </span>
                  <svg
                    class="size-4 transition "
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                    />
                  </svg>
                </button>
              </header>
            </article>
            <article
              class="rounded-t-sm bg-[#222931]"
            >
              <header>
                <button
                  class="w-full flex justify-between items-center px-4 py-2 border border-white/10"
                  type="button"
                >
                  <span
                    class="uppercase"
                  >
                    Ramp chart
                  </span>
                  <svg
                    class="size-4 transition "
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                    />
                  </svg>
                </button>
              </header>
            </article>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-4 bg-[#15191E]"
    >
      <div
        class="px-4"
      >
        <div
          class="flex gap-4 justify-between items-center font-light"
        >
          <div>
            <h4
              class="ml-1 text-sm"
            >
              Melbourne
            </h4>
            <span
              class="text-5xl ml-3"
            >
              32
              °
            </span>
            <div
              class="flex text-xs gap-2"
            >
              <span>
                Sun 7th
              </span>
              <span>
                9:29 PM
              </span>
            </div>
          </div>
          <div>
            <svg
              class="size-20"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M440.1 355.2l-39.2-23 34.1-9.3c8.4-2.3 13.4-11.1 11.1-19.6l-4.1-15.5c-2.2-8.5-10.9-13.6-19.3-11.3L343 298.2 271.2 256l71.9-42.2 79.7 21.7c8.4 2.3 17-2.8 19.3-11.3l4.1-15.5c2.2-8.5-2.7-17.3-11.1-19.6l-34.1-9.3 39.2-23c7.5-4.4 10.1-14.2 5.8-21.9l-7.9-13.9c-4.3-7.7-14-10.3-21.5-5.9l-39.2 23 9.1-34.7c2.2-8.5-2.7-17.3-11.1-19.6l-15.2-4.1c-8.4-2.3-17 2.8-19.3 11.3l-21.3 81-71.9 42.2v-84.5L306 70.4c6.1-6.2 6.1-16.4 0-22.6l-11.1-11.3c-6.1-6.2-16.1-6.2-22.2 0l-24.9 25.4V16c0-8.8-7-16-15.7-16h-15.7c-8.7 0-15.7 7.2-15.7 16v46.1l-24.9-25.4c-6.1-6.2-16.1-6.2-22.2 0L142.1 48c-6.1 6.2-6.1 16.4 0 22.6l58.3 59.3v84.5l-71.9-42.2-21.3-81c-2.2-8.5-10.9-13.6-19.3-11.3L72.7 84c-8.4 2.3-13.4 11.1-11.1 19.6l9.1 34.7-39.2-23c-7.5-4.4-17.1-1.8-21.5 5.9l-7.9 13.9c-4.3 7.7-1.8 17.4 5.8 21.9l39.2 23-34.1 9.1c-8.4 2.3-13.4 11.1-11.1 19.6L6 224.2c2.2 8.5 10.9 13.6 19.3 11.3l79.7-21.7 71.9 42.2-71.9 42.2-79.7-21.7c-8.4-2.3-17 2.8-19.3 11.3l-4.1 15.5c-2.2 8.5 2.7 17.3 11.1 19.6l34.1 9.3-39.2 23c-7.5 4.4-10.1 14.2-5.8 21.9L10 391c4.3 7.7 14 10.3 21.5 5.9l39.2-23-9.1 34.7c-2.2 8.5 2.7 17.3 11.1 19.6l15.2 4.1c8.4 2.3 17-2.8 19.3-11.3l21.3-81 71.9-42.2v84.5l-58.3 59.3c-6.1 6.2-6.1 16.4 0 22.6l11.1 11.3c6.1 6.2 16.1 6.2 22.2 0l24.9-25.4V496c0 8.8 7 16 15.7 16h15.7c8.7 0 15.7-7.2 15.7-16v-46.1l24.9 25.4c6.1 6.2 16.1 6.2 22.2 0l11.1-11.3c6.1-6.2 6.1-16.4 0-22.6l-58.3-59.3v-84.5l71.9 42.2 21.3 81c2.2 8.5 10.9 13.6 19.3 11.3L375 428c8.4-2.3 13.4-11.1 11.1-19.6l-9.1-34.7 39.2 23c7.5 4.4 17.1 1.8 21.5-5.9l7.9-13.9c4.6-7.5 2.1-17.3-5.5-21.7z"
              />
            </svg>
          </div>
        </div>
        <div
          class="mt-5"
        >
          <ul
            class="space-y-0.5"
          >
            <li
              class="flex gap-5 text-sm items-center"
            >
              <span
                class="font-light flex-1"
              >
                Humidity
              </span>
              <span
                class="flex gap-0.5 items-center w-1/3"
              >
                <span
                  class="font-medium text-base"
                >
                  78%
                </span>
              </span>
            </li>
            <li
              class="flex gap-5 text-sm items-center"
            >
              <span
                class="font-light flex-1"
              >
                Chance of Rain
              </span>
              <span
                class="flex gap-0.5 items-center w-1/3"
              >
                <span
                  class="font-medium text-base"
                >
                  34%
                </span>
              </span>
            </li>
            <li
              class="flex gap-5 text-sm items-center"
            >
              <span
                class="font-light flex-1"
              >
                Wind
              </span>
              <span
                class="flex gap-0.5 items-center w-1/3"
              >
                <span
                  class="font-medium text-base"
                >
                  21
                </span>
                <span
                  class="font-light"
                >
                  kmh
                </span>
              </span>
            </li>
            <li
              class="flex gap-5 text-sm items-center"
            >
              <span
                class="font-light flex-1"
              >
                Tomorrow
              </span>
              <span
                class="flex gap-0.5 items-center w-1/3"
              >
                <span
                  class="font-medium text-base"
                >
                  30°
                </span>
                <svg
                  class="size-4"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
                  />
                </svg>
              </span>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="mt-8"
      >
        <section
          class="space-y-4"
        >
          <article
            class="rounded-t-sm bg-[#222931]"
          >
            <header>
              <button
                class="w-full flex justify-between items-center px-4 py-2 border border-white/10"
                type="button"
              >
                <span
                  class="uppercase"
                >
                  Delayed Routes
                </span>
                <svg
                  class="size-4 transition "
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                  />
                </svg>
              </button>
            </header>
          </article>
          <article
            class="rounded-t-sm bg-[#222931]"
          >
            <header>
              <button
                class="w-full flex justify-between items-center px-4 py-2 border border-white/10"
                type="button"
              >
                <span
                  class="uppercase"
                >
                  Ramp chart
                </span>
                <svg
                  class="size-4 transition "
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                  />
                </svg>
              </button>
            </header>
          </article>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
